source/game/system/map/MapdataCheckPoint.cc

```diff
@@ -137,9 +137,59 @@ void MapdataCheckPoint::clearSearched() {
     m_searched = false;
 }
 
+/// @addr{0x80511EC8}
+/// @brief Finds the offset between the two positions that enter the checkpoint.
+/// @details This assumes the player is entering the checkpoint as intended, and not from the side.
+/// @param prevPos The previous position, likely not located in the checkpoint.
+/// @param pos The current position, likely located in the checkpoint.
+/// @return The earliest subdivision that crosses into the checkpoint, in the range [1, 17].
+u16 MapdataCheckPoint::getEntryOffsetMs(const EGG::Vector2f &prevPos,
+        const EGG::Vector2f &pos) const {
+    constexpr f32 REFRESH_PERIOD = 1000.0f / 59.94f;
```

Comment:
Redefinition from `TimerManager::calc()`. If you ever foresee this being re-used in other functions as well, then I think we should move this to some common file, either `Common.hh` or a new header that's specific to `namespace System`

Reply:
This will not be used elsewhere.